contracts/airdrops/ExistingTokenDrop.sol

```diff
@@ -64,59 +48,52 @@ contract ExistingTokenDrop is AirdropInfo, AirdropMerkleProof, Ownable {
         airdropDuration = _airdropDuration * 1 days;
         airdropStartTime = block.timestamp;
         airdropFinishTime = block.timestamp + airdropDuration;
-        airBroFactoryAddress = _airBroFactoryAddress;
-    }
-
-    /// @notice Sets the merkleRoot - can only be done if admin (different from the contract owner)
-    /// @param _merkleRoot - The root hash of the Merle Tree
-    function setMerkleRoot(bytes32 _merkleRoot) external onlyAdmin {
-        merkleRoot = _merkleRoot;
-        emit MerkleRootChanged(_merkleRoot);
     }
 
     /// @notice Allows the airdrop creator to provide funds for airdrop reward
     function fundAirdrop() external {
         if (rewardToken.balanceOf(msg.sender) < totalAirdropAmount) revert InsufficientAmount();
         if (airdropFunded) revert AlreadyFunded();
-        rewardToken.transferFrom(msg.sender, address(this), totalAirdropAmount);
+
         airdropFunded = true;
         airdropFundBlockTimestamp = block.timestamp;
         airdropFundingHolder = msg.sender;
-        emit AirdropFunded();
+
+        rewardToken.transferFrom(msg.sender, address(this), totalAirdropAmount);
```

Comment:
Use SafeERC20 for ERC20 tokens that you don't control. Some of them don't revert, instead they return `false` so in that case, the token transfer would fail, but this would not revert making the contract useless, because it would be considered funded, and it would contain no token